sandbox: allow retaining capabilities

ripatel-fd opened the pull request.

mmcgee-jump commented on src/util/sandbox/fd_sandbox.c:
```diff
@@ -444,7 +445,14 @@ fd_sandbox_private_drop_caps( ulong cap_last_cap ) {
+  data[ 1 ].permitted &= (uint)( desired_capabilities>>32 );
+  data[ 0 ].effective = data[ 0 ].permitted;
+  data[ 1 ].effective = data[ 1 ].permitted;
+  data[ 0 ].inheritable = 0U;
```
These are already 0?

mmcgee-jump commented on src/util/sandbox/fd_sandbox.c:
```diff
@@ -444,7 +445,14 @@ fd_sandbox_private_drop_caps( ulong cap_last_cap ) {
   struct __user_cap_header_struct hdr = { _LINUX_CAPABILITY_VERSION_3, 0 };
   struct __user_cap_data_struct   data[2] = { { 0 } };
-  if( -1==syscall( SYS_capset, &hdr, data ) )                          FD_LOG_ERR(( "syscall(SYS_capset) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+  if( -1==syscall( SYS_capget, &hdr, data ) ) FD_LOG_ERR(( "syscall(SYS_capget) failed (%i-%s)", errno, fd_io_strerror( errno ) ));
```
Why do you capget? The point is to clear the caps, you can just set it to keep the `desired` set, no need to keep the other stuff

mmcgee-jump reviewed (changes requested):
.

riptl committed:
Netlink tile bug fixes

- Fix overrun recovery
- Fix dispatching of neighbor solicit requests
- Log warning for neighbor discovery failures

mmcgee-jump commented on src/disco/stem/fd_stem.c:
```diff
@@ -554,6 +554,11 @@ STEM_(run1)( ulong                        in_cnt,
         this_in->accum[ FD_METRICS_COUNTER_LINK_OVERRUN_POLLING_FRAG_COUNT_OFF ] += (uint)(-diff);
       }
+
+#ifdef STEM_CALLBACK_AFTER_POLL_OVERRUN
```
Wrong place, needs to be in the if statement above?

mmcgee-jump replied:
Also need to document this, see the top of the file

riptl committed:
sandbox: allow retaining capabilities

riptl committed:
Grant CAP_NET_ADMIN to netlink tile
